File: rustHttpStaticServer/src/HttpParser/httpParser.rs
enum HttpMethod{
    GET,
    HEAD,
    POST,
    PUT,
    DELETE,
    CONNECT,
    OPTIONS,
    TRACE,
    PATCH,
}

struct HttpRequest{
    method: HttpMethod,
    version:  &str,
    route:  &str,
    headers: HashMap<&str, &str>
}

fn getHttpMethod(m: &str) -> HttpMethod {
    return match m{
        "GET" => HttpMethod.GET,
        "HEAD" => HttpMethod.HEAD,
        "POST" => HttpMethod.POST,
        "PUT" => HttpMethod.PUT,
        "DELETE" => HttpMethod.DELETE,
        "OPTIONS" => HttpMethod.OPTIONS,
        "TRACE" => HttpMethod.TRACE,
        "PATCH" => HttpMethod.PATCH
    }
}

fn parseFirstLine(line: &str) -> HttpRequest{
    let method: HttpMethod;
    let version: &str;
    let route:  &str;

}

File: HttpServerInC/httpParser/http_parser.c
#include <stdio.h>
#include <stdlib.h>

struct HttpHeader{
    char *name;
    char *value;
};


enum HttpMethod{
    GET,
    HEAD,
    POST,
    PUT,
    DELETE,
    CONNECT,
    OPTIONS,
    TRACE,
    PATCH,
};


struct HttpRequest{
    enum HttpMethod method;
    char* version;
    struct HttpHeader *httpHeaders;
    size_t httpHeadersSize;
    unsigned char *body 
};

struct HttpRequest* parse_first_line(){

}